templates/commands/render_action.go

```diff
@@ -100,3 +111,150 @@ func templateAndCompileRegexes(regexes []model.String, inputs map[string]string)
 
 	return compiled, merr
 }
+
+// A template parser helper to remove the boilerplate of parsing with our
+// desired options.
+func parseGoTmpl(tpl string) (*template.Template, error) {
+	return template.New("").Option("missingkey=error").Parse(tpl) //nolint:wrapcheck
+}
+
+// pos may be nil if the template is not coming from the spec file and therefore
+// there's no reason to print out spec file location in an error message. If
+// template execution fails because of a missing input variable, the error will
+// be wrapped in a unknownTemplateKeyError.
+func parseAndExecuteGoTmpl(pos *model.ConfigPos, tmpl string, inputs map[string]string) (string, error) {
+	parsedTmpl, err := parseGoTmpl(tmpl)
+	if err != nil {
+		return "", model.ErrWithPos(pos, `error compiling as go-template: %w`, err) //nolint:wrapcheck
+	}
+
+	// As of go1.20, if the template references a nonexistent variable, then the
```

Comment:
Optional side quest: do you want to file a bug upstream about this?

Reply:
I just found https://github.com/golang/go/issues/57801 for this, and added my thumbs up emoji to it 😀